src/JustSaying/AwsTools/MessageHandling/Dispatch/MessageDispatcher.cs

```diff
@@ -90,7 +90,12 @@ await middleware.RunAsync(handleContext, null, cancellationToken)
 
             return (false, null, null);
         }
-        catch (Exception ex)
+        catch (OperationCanceledException)
+        {
+            // Ignore cancellation
+            return (false, null, null);
+        }
+        catch (Exception ex) when (ex is not OperationCanceledException)
```

Comment:
Is the `when` needed? I thought it would be excluded by the fact it's caught above?